packages/theme/src/types.ts

```diff
@@ -25,16 +25,54 @@ export type ThemeBreakpoints = {
     [key: string]: string;
 };
 
+/*
+ * Typography section
+ */
+export type TypographyType = "headings" | "paragraphs" | "quotes" | "lists" | string;
+export type TypographyStyle = {
+    id: string;
+    name: string;
+    tag: string;
+    styles: CSSObject;
+};
+
+export type Typography = Record<TypographyType, Readonly<TypographyStyle[]>>;
+export type ThemeTypographyStyleItems = TypographyStyle[];
+
 export interface ThemeStyles {
     colors: Record<string, any>;
     borderRadius?: number;
-    typography: Record<string, StylesObject>;
+    typography: Typography;
     elements: Record<string, Record<string, any> | StylesObject>;
+    [key: string]: any;
+}
+
+/*
+ * Decorated typography types
+ */
+
+export type DecoratedThemeTypographyStyles = ThemeTypographyStyleItems & {
+    stylesById: (id: string) => CSSObject | undefined;
+};
+
+export type DecoratedTypography = Record<TypographyType, DecoratedThemeTypographyStyles>;
 
+interface DecoratedThemeStyles extends Omit<ThemeStyles, "typography"> {
+    colors: Record<string, any>;
+    borderRadius?: number;
+    typography: DecoratedTypography;
+    elements: Record<string, Record<string, any> | StylesObject>;
     [key: string]: any;
 }
 
-export interface Theme {
+export interface BaseTheme {
     breakpoints: ThemeBreakpoints;
     styles: ThemeStyles;
 }
+
+export interface DecoratedTheme {
```

Comment:
Let's not export `DecoratedTheme`, just so it doesn't confuse the users or the users accidentally don't use this. Let's make sure they only use `Theme`. Also, with this approach, we are sure nobody used `DecoratedTheme` and we can safely later remove it.

Reply:
done